internal/pkg/ocisif/datacontainer.go

```diff
@@ -158,3 +167,68 @@ func DataContainerLayerOffset(f *os.File) (int64, error) {
 	offset, err := layers[0].(*ocitsif.Layer).Offset()
 	return offset, err
 }
+
+// oci11Artifact adapts the base image to comply with the OCI v1.1 artifact specification.
+type oci11Artifact struct {
+	v1.Image
+}
+
+// Size returns the size of the manifest.
+func (w *oci11Artifact) Size() (int64, error) {
+	mf, err := w.RawManifest()
+	if err != nil {
+		return 0, err
+	}
+
+	return int64(len(mf)), nil
+}
+
+// Digest returns the sha256 of this image's manifest.
+func (w *oci11Artifact) Digest() (v1.Hash, error) {
+	mf, err := w.RawManifest()
+	if err != nil {
+		return v1.Hash{}, err
+	}
+
+	h, _, err := v1.SHA256(bytes.NewReader(mf))
+	if err != nil {
+		return v1.Hash{}, err
+	}
+	return h, nil
+}
+
+// RawManifest returns the serialized bytes of Manifest().
+func (w *oci11Artifact) RawManifest() ([]byte, error) {
+	mf, err := w.Image.RawManifest()
+	if err != nil {
+		return nil, err
+	}
+
+	var manifest struct {
+		SchemaVersion int64             `json:"schemaVersion"`
+		MediaType     types.MediaType   `json:"mediaType,omitempty"`
+		ArtifactType  string            `json:"artifactType,omitempty"`
+		Config        v1.Descriptor     `json:"config"`
+		Layers        []v1.Descriptor   `json:"layers"`
+		Annotations   map[string]string `json:"annotations,omitempty"`
+		Subject       *v1.Descriptor    `json:"subject,omitempty"`
+	}
```

Comment:
Can't simplify this a bit with?

```suggestion
	var manifest struct {
		v1.Manifest
		ArtifactType  string            `json:"artifactType,omitempty"`
	}
```

Reply:
That's definitely simpler, but I think that will change the order of the fields? That doesn't matter at the JSON level, but that would mean that we'd generate a different manifest digest than GGCR likely will (if/when https://github.com/google/go-containerregistry/pull/1931 lands.)

Reply:
That makes sense